internal/cli/up.go

```diff
@@ -95,6 +100,8 @@ local resources--i.e. those using serve_cmd--are terminated when you exit Tilt.
 	cmd.PreRun = func(cmd *cobra.Command, args []string) {
 		c.hudFlagExplicitlySet = cmd.Flag("hud").Changed
 		c.watchFlagExplicitlySet = cmd.Flag("watch").Changed
+		c.legacyFlagExplicitlySet = cmd.Flag("legacy").Changed
```

Comment:
i understand why you copied this from the above flags. but this pattern should really only be used for new flags where we're trying to preserve some backwards-compatibility. it never makes sense to read `Changed` on newly introduced flags

Reply:
@nicks I'm taking a second look at this, and I'm slightly confused.
Wouldn't I need this boolean to see if the user has explicitly invoked the --legacy or --stream flags?